src/controllers/shopping-cart.controller.ts

```diff
@@ -78,6 +85,14 @@ export class ShoppingCartController {
     @param.path.string('userId') userId: string,
     @requestBody({description: 'shopping cart item'}) item: ShoppingCartItem,
   ) {
-    await this.shoppingCartRepository.addItem(userId, item);
+    debug('Add item %j to shopping cart %s', item, userId);
+    return retry<ShoppingCart>(
+      {
+        run: () => this.shoppingCartRepository.addItem(userId, item),
+        description: `update the shopping cart for '${userId}'`,
+      },
+      10,
+      100,
```

Comment:
I find these positional arguments difficult to understand. What is `10` and `100`?

I am proposing to change `retry` API to accept an options object instead of positional parameters.

```ts
    return retry<ShoppingCart>(
       {
         run: () => this.shoppingCartRepository.addItem(userId, item),
         description: `update the shopping cart for '${userId}'`,
       },
       {
         maxRetries: 10,
         retryInterval: 100, 
       });
```